lib/common/memory.c

```diff
@@ -152,27 +153,26 @@ void *h2o_mem_alloc_pool(h2o_mem_pool_t *pool, size_t sz)
     }
 
     /* return a valid pointer even for 0 sized allocs */
-    if (sz == 0)
+    if (H2O_UNLIKELY(sz == 0))
         sz = 1;
 
-    /* 16-bytes rounding */
-    sz = (sz + 15) & ~15;
-    if (sizeof(pool->chunks->bytes) - pool->chunk_offset < sz) {
+    if (sizeof(*pool->chunks) - pool->chunk_offset < sz) {
         /* allocate new chunk */
-        struct st_h2o_mem_pool_chunk_t *newp = h2o_mem_alloc_recycle(&mempool_allocator, sizeof(*newp));
+        union un_h2o_mem_pool_chunk_t *newp = h2o_mem_alloc_recycle(&mempool_allocator, sizeof(*newp));
         newp->next = pool->chunks;
         pool->chunks = newp;
-        pool->chunk_offset = 0;
+        pool->chunk_offset = sizeof(union un_h2o_mem_pool_chunk_t *);
```

Comment:
Can we change the sizeof to `sizeof(newp->next)`?

Relying on the variable name is more prone to errors than copy-pasting the type of the variable name.

Reply:
yes, will be fixed.